bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/service/ReadyUtil.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.core.service;
+
+import java.util.Dictionary;
+import java.util.Properties;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
+import org.osgi.framework.ServiceRegistration;
+
+/**
+ * Utility class to easily register {@link ReadyMarker}s programmatically.
+ *
+ * @author Simon Kaufmann - initial contribution and API
+ *
+ */
+@SuppressWarnings({ "rawtypes", "unchecked" })
+public class ReadyUtil {
```

Comment:
IIRC we already discussed the "omit supress warnings for the whole class if possible" already.

Isn't it possible here?